juce_serialport.h

```diff
@@ -190,7 +190,7 @@ class JUCE_API SerialPortInputStream : public juce::InputStream, public juce::Ch
 		const juce::ScopedLock l(bufferCriticalSection);
 		return bufferedbytes?false:true;
 	};
-	virtual juce::int64 getPosition(){return -1;}
+	virtual juce::int64 getPosition(){return 0;}
```

Comment:
While the JUCE API isn't clear on how to derive streams which do not support getPosition, it seems that returning a -1 is a good way to indicate the lack of support. I would even suggest that is why a signed return type was chosen.

Reply:
As I mentioned, the problem is that in `juce_core/streams/juce_InputStream.cpp` there's this:
```
int64 InputStream::getNumBytesRemaining()
{
    auto len = getTotalLength();

    if (len >= 0)
        len -= getPosition();

    return len;
}

Now, when the buffer is empty, `getTotalLength()` returns zero, `getPosition()` returns -1, and therefore `getNumBytesRemaining()` returns 1 for an empty stream, which is certainly wrong? And in general, it will return one too many. And it's not virtual, so we can't simply override it.

So while I agree that per se returning -1 is an established pattern, here the fact that it breaks checking the amount of bytes available is worse than just returning 0.

While

Reply:
On the other hand, if we want to go deeper into the rabbit hole: from `juce_InputStream.h`
```
//==============================================================================
    /** Returns the total number of bytes available for reading in this stream.

        Note that this is the number of bytes available from the start of the
        stream, not from the current position.

        If the size of the stream isn't actually known, this will return -1.

        @see getNumBytesRemaining
    */
    virtual int64 getTotalLength() = 0;

    /** Returns the number of bytes available for reading, or a negative value if
        the remaining length is not known.
        @see getTotalLength
    */
    int64 getNumBytesRemaining();
```
This actually suggests that `getTotalLength()` should also always return -1, in which case `getNumBytesRemaining()`, will do likewise, and in which case there's indeed no further harm in returning -1 from `getPosition()` too.

In that case, I guess the right way to check if I can read a byte would be to check `isExhausted()`?

Reply:
So to sum up, I'd suggest either
 1) keep it as per my PR, i.e. `getPosition()` returns 0. Upside is that then it's possible to check how many bytes are currently buffered, downside is that, as you implied, it's actually inconsistent with what's written in the Juce docs, and it makes it look like position would be supported.

or

 2) go back to `getPosition()` returns -1, and also make `getTotalLength()` return -1. The upside is that that's actually consistent with the juce header comments, the downside is that then it won't be possible (unless I'm missing something?) to check how many bytes are available for reading.

Of course I'm also open to being convinced for some option 3).

Reply:
As you know I didn't write the library, and so I only learn about the internals when I need to look, and it  seems like `getTotalLength()` is doing the wrong thing, in that we can't really know the length of the stream. I'm torn between either change, as they are both possibly breaking changes for people who use those APIs. But, having said that, the least harmful break is in modifying `getPostion()` to return 0, in that no one could have relied on the return value before, and based on the error in `getTotalLength()`, no one could have relied on that return value either. So, let's keep your change! :)